frontend-react/src/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from "@trussworks/react-uswds";
```

Comment:
This is something that @etanb and I chatted about yesterday, but I think we're going to have to start thinking about our directory structure for more scalability.  Specifically for atomic components like this, it'd be great to put them in a `core/` or `common/` directory so that we know they're either very widely used throughout our site or wrapping/replacing Trussworks' components.

Off the top of my head, my feeling is that we can start making "domains" (from domain-driven design) that'll have a structure like:

```
components/  # where all components for this domain live
tests/               # where all tests for this domain live
utils/                # where all utility functions/classes for this domain live
hooks/             # where all hooks for this domain live
```

For our use case, I'd imagine it to be something like:
```
frontend-react/
  src/
    --- new domain --- 
    core/
      components/
        IconButton.tsx
      tests/
        IconButton.test.tsx
        DateTimeUtils.test.ts
      utils/
        DateTimeUtils.ts
      hooks/

    --- new domain --- 
    analytics/
      components/
      tests/
        Analytics.test.ts
      utils/
        Analytics.ts
      hooks/
        UseCustomTrackEvent.ts # or whatever if we'd ever need that 🤷

    --- new domain ---
    organizations/ # idk maybe?
```

What do you think about this?  I can also bring it up in Slack so we can chat more about it as a team!

Reply:
I'm torn between a component folder per component containing all their relevant files (component, test, storybook story, etc.) or those types of files having their own dedicated folder. Both makes sense, but I do like having a component folder that I know I can go directly to get all relevant just for that specifically.

Reply:
For clarity, this:

`frontend-react/
    src/
        core/
            components/
                IconButton/
                    IconButton.tsx
                    IconButton.test.tsx
                    IconButton.story.tsx
...`

Reply:
Sure, that sounds good to me!  I don't have a strong preference of how domains are organized internally (as long as they're structured well) -- just that we have proper separation of concerns in the filesystem!

How about we hold off on directory reorganization for now and bring it up during our Experience eng sync on Thursday to get everyone involved?  If we decide on a more dramatic change like this, we can use the new pattern for any new components and do a gradual migration for existing ones.  What do you think?

Reply:
Absolutely. Would you like me to fold IconButton into the components folder directly for consistency in the current norm?

Reply:
Ah, good point!  Yeah, if you wouldn't mind!  I think that would be good just so it'll be more consistent as we take on a possible reorganization in the future 🙏